lnwallet/channel_test.go

```diff
@@ -7812,3 +7811,385 @@ func TestFetchParent(t *testing.T) {
 		})
 	}
 }
+
+// TestEvaluateView tests the creation of a htlc view and the opt in mutation of
+// send and receive balances. This test does not check htlc mutation on a htlc
+// level.
+func TestEvaluateView(t *testing.T) {
+	const (
+		// nextHeight is a constant that we use for the next height in
+		// all unit tests.
+		nextHeight = 400
+
+		// feePerKw is the fee we start all of our unit tests with.
+		feePerKw = 1
+
+		// htlcAddAmount is the amount for htlc adds in tests.
+		htlcAddAmount = 15
+
+		// ourFeeUpdateAmt is an amount that we update fees to
+		// expressed in msat.
+		ourFeeUpdateAmt = 20000
+
+		// ourFeeUpdatePerSat is the fee rate *in satoshis* that we
+		// expect if we update to ourFeeUpdateAmt.
+		ourFeeUpdatePerSat = chainfee.SatPerKWeight(20)
+
+		// theirFeeUpdateAmt iis an amount that they update fees to
+		// expressed in msat.
+		theirFeeUpdateAmt = 10000
+
+		// theirFeeUpdatePerSat is the fee rate *in satoshis* that we
+		// expect if we update to ourFeeUpdateAmt.
+		theirFeeUpdatePerSat = chainfee.SatPerKWeight(10)
+	)
+
+	// fetchParentFromMap takes a map of htlc index to htlc parent indexes
+	// and returns a fetch parent function which returns a payment
+	// descriptor with that parent index set as the htlc add index. If the
+	// child htlc is not a key in the map, this function will error.
+	fetchParentFromMap := func(parentMap map[uint64]uint64) fetchParent {
```

Comment:
This looks risky to me. To replace such a low-level component of the system under test. Isn't it testable using the real `fetchParent`?

Reply:
Good point, replaced with the real thing.